recipe/mpiexec.sh

```diff
@@ -1,4 +1,16 @@
 #!/bin/bash
 set -e
+
+MPI_VARIANT=mpich
```

Comment:
I believe that the `mpi` env variable will be defined here, so we can use that for the switch.

Reply:
OK, I did not know, I used an alternative approach in my previous PRs (checking for mpichversion and ompi_info). Now I realize these checks are quite good as implemented (though they work just fine in the CI build context).

In the long run we need a more scalable way to handle the `mpiexec` issues. I'm just tired of coping over the mpiexec.sh script on every other feedstock.

Maybe we should create a new package `mpi-toolchain` ?

Reply:
Yes, the checks should use $`PREFIX/<program>`, dumb me. And yes, I think we should add them in mpich/openmpi eventually (under `$PREFIX/bin`), and name them something ugly like `conda-forge-ci-mpiexec.sh`. Would that work for you?

Reply:
Yeah, I think so. If it's truly private, maybe not even in `/bin`, but `libexec` or `share`